doc/source/architecture.rst

```diff
@@ -0,0 +1,188 @@
+.. _ref_architecture:
+
+=========================
+PyMechanical Architecture
+=========================
+
+PyMechanical is a Python interface to the Mechanical application. Mechanical
+is a polyglot [#f1]_ desktop application whose graphical user interface (GUI)
+can run on both Windows and Linux.
+
+For several years now, Mechanical has had Python scripting capabilities using
+a scripting API. Like many other applications developed by Ansys, scripting
+became heavily used for the purpose of *automation*. A user who  would spend
+time doing the same thing over and over again with the GUI could instead write
+a script to do the same thing, saving time and effort.
+
+Also like many other applications, Mechanical's API served another purpose.
+That is, it was used for *customization*. Users could use the API as a way to
+add capabilities to Mechanical. It has been used to add buttons to the User
+Interface, add custom objects to the data model, even as far as adding
+third-party solvers that can take advantage of the powerful and intuitive pre-
+and post-processing capabilities of Mechanical.
+
```

Comment:
You can also use APIs to customize your workflows by creating customized options, objects, as well as to incorporate third-party solvers in order to enhance pre- and post-processing capabilities.